servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -0,0 +1,451 @@
+    mLockManager.cleanupUser(userId);
+    mEvictionLock.readLock().unlock();
+
+    // TODO: fix the block removing below,
+    Set<String> dirs = new HashSet<String>(tempBlocksToRemove.size());
+    for (TempBlockMeta tempBlockMeta : tempBlocksToRemove) {
+      String fileName = tempBlockMeta.getPath();
+      try {
+        String dirName = CommonUtils.getParent(fileName);
+        dirs.add(dirName);
+      } catch (InvalidPathException e) {
+        LOG.error("Error in cleanup userId {}: cannot parse parent dir of {}", userId, fileName);
+      }
+      if (!new File(fileName).delete()) {
+        LOG.error("Error in cleanup userId {}: cannot delete file {}", userId, fileName);
+      }
+    }
+    // Cleanup the user folder
+    for (String dirName : dirs) {
+      if (!new File(dirName).delete()) {
+        LOG.error("Error in cleanup userId {}: cannot delete directory ", userId, dirName);
+      }
+    }
+  }
+
+  @Override
+  public BlockStoreMeta getBlockStoreMeta() {
+    return mMetaManager.getBlockStoreMeta();
+  }
+
+  @Override
+  public void registerBlockStoreEventListener(BlockStoreEventListener listener) {
+    synchronized (mBlockStoreEventListeners) {
+      mBlockStoreEventListeners.add(listener);
+    }
+  }
+
+  // Create a temp block meta. This method requires {@link mEvictionLock} is acquired in read mode.
+  private TempBlockMeta createBlockMetaNoLock(long userId, long blockId,
+      BlockStoreLocation location, long initialBlockSize) throws IOException {
+    if (mMetaManager.hasTempBlockMeta(blockId)) {
+      throw new IOException("Failed to create TempBlockMeta: blockId " + blockId + " exists");
+    }
+    if (mMetaManager.hasBlockMeta(blockId)) {
+      throw new IOException("Failed to create TempBlockMeta: blockId " + blockId + " committed");
+    }
+    TempBlockMeta tempBlock = mAllocator.allocateBlock(userId, blockId, initialBlockSize, location);
+    if (tempBlock == null) {
+      // Failed to allocate a temp block, let Evictor kick in to ensure sufficient space available.
+
+      // Upgrade to write lock to guard evictor. This is not a real "upgrade" and things can
+      // happen between unlock and lock.
+      mEvictionLock.readLock().unlock();
+      mEvictionLock.writeLock().lock();
+      try {
+        freeSpaceInternal(userId, initialBlockSize, location);
+      } finally {
+        // Downgrade to read lock again after eviction. Lock before unlock is intentional.
+        mEvictionLock.readLock().lock();
+        mEvictionLock.writeLock().unlock();
+      }
+      tempBlock = mAllocator.allocateBlock(userId, blockId, initialBlockSize, location);
+      Preconditions.checkNotNull(tempBlock, "Cannot allocate block %s:", blockId);
+    }
+    // Add allocated temp block to metadata manager
+    mMetaManager.addTempBlockMeta(tempBlock);
+    return tempBlock;
+  }
+
+  // Commit a temp block. This method requires no eviction lock acquired.
```

Comment:
This comment is confusing. The evictionLock should already be locked before calling this method, right? The comment sounds like you don't need the lock to call this method...